presto-tests/src/main/java/com/facebook/presto/tests/FeatureSelection.java

```diff
@@ -0,0 +1,137 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class FeatureSelection
+{
+    public static Builder all()
+    {
+        return new Builder(Optional.empty());
+    }
+
+    public static Builder includes(Feature... features)
+    {
+        return includes(ImmutableSet.copyOf(features));
+    }
+
+    public static Builder includes(Collection<Feature> features)
+    {
+        requireNonNull(features);
+        return new Builder(Optional.of(ImmutableSet.copyOf(features)));
+    }
+
+    public enum Feature
+    {
+        ENGINE_ONLY,
+        JOIN,
+        GROUP_BY,
+        SORT,
+        SPILL,
+        VIEW;
+    }
+
+    private final Optional<Set<Feature>> includes;
+    private final Optional<Set<Feature>> excludes;
+
+    private FeatureSelection(Optional<Set<Feature>> includes, Optional<Set<Feature>> excludes)
+    {
+        requireNonNull(includes, "includes is null");
+        requireNonNull(excludes, "excludes is null");
+
+        if (includes.isPresent()) {
+            this.includes = Optional.of(ImmutableSet.copyOf(includes.get()));
+        }
+        else {
+            this.includes = Optional.empty();
+        }
+        if (excludes.isPresent()) {
+            this.excludes = Optional.of(ImmutableSet.copyOf(excludes.get()));
+        }
+        else {
+            this.excludes = Optional.empty();
+        }
+    }
+
+    public boolean isSelected(Feature feature)
+    {
+        requireNonNull(feature);
+        return areSelected(ImmutableSet.of(feature));
+    }
+
+    public boolean areSelected(Set<Feature> features)
+    {
+        requireNonNull(features);
+
+        if (excludes.isPresent() && excludes.get().stream().anyMatch(features::contains)) {
+            return false;
+        }
+
+        if (includes.isPresent() && includes.get().stream().anyMatch(features::contains)) {
+            return true;
+        }
+
+        return !includes.isPresent();
+    }
+
+    public static class Builder
+    {
+        private final Optional<Set<Feature>> includes;
+        private Optional<Set<Feature>> excludes = Optional.empty();
+
+        private Builder(Optional<Set<Feature>> includes)
+        {
+            this.includes = requireNonNull(includes, "includes is null");
+        }
+
+        public Builder without(Feature... features)
```

Comment:
maybe just keep only this method for simplicity and to be symmetrical with `includes` signature.

Also this method could just return `FeatureSelection`. Then we could just write:
```
FeatureSelection.includes(...).without(...);
```

Reply:
Rename to `excludes` (or better `excluding`) to be symmetrical with `includes` (`including`)?

Or maybe rename `includes -> features` and `all -> allFeatures`. Then we could just write (using static import):
```
features(...).excluding(...);
allFeatures().excluding(...);
```